python/lancedb/table.py

```diff
@@ -523,23 +524,109 @@ def update(
         raise NotImplementedError
 
 
+class LanceTableMode:
+    pass
+
+
+@dataclass
+class LanceStandardMode(LanceTableMode):
+    last_consistency_check: Optional[float] = None
+
+
+@dataclass
+class LanceTimeTravelMode(LanceTableMode):
+    version: int
+
+
 class LanceTable(Table):
     """
     A table in a LanceDB database.
+
+    This can be opened in two modes: standard and time-travel.
+
+    Standard mode is the default. In this mode, the table is mutable and tracks
```

Comment:
Should these docs be here or with the data classes?
The users don't actually set the mode, they just set the version number. So they might get confused? It might make more sense here to just say if you don't pass in a version number then the table is mutable and tracks the latest version etc?

Reply:
> The users don't actually set the mode, they just set the version number

They do not. For the users, the modes are just naming concepts. That makes it easier to talk about them and explain.

>  So they might get confused?

I was trying to make it less confusing by given them a conceptual model for the two "modes" that it can run in. But I can leave the two modes unnamed instead, if you think that's clearer. We can just call it, "the mode where you passed an explicit version or called checkout()" and "the mode where you didn't pass an explicit version or call checkout()".